### ISSUE TITLE
`no-extra-parens` has false negative on octal numeric literals.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#7122]
**Tell us about your environment**

* **ESLint Version:** 3.5.0
* **Node Version:** n/a - confirmed in our online demo. (Chrome 54)
* **npm Version:** n/a

**What parser (default, Babel-ESLint, etc.) are you using?**

- default

**Please show your full configuration:**

- nothing

**What did you do? Please include the actual source code causing the issue.**

Lint the following code on the online demo.

```js
/*eslint no-extra-parens:error*/

(0123).toString()
```

Or

```bash
$ echo "(0123).toString()" | eslint --stdin --no-eslintrc --rule no-extra-parens:error
```

**What did you expect to happen?**

1 error.
The parentheses which are enclosing the octal numeric literal are unnecessary.
`0123.toString()` is legal since `OctalIntegerLiteral` production never ends with `.`.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors.

**Comment by mysticatea** [comment]
Note: Requires `acorn@4.0.2` to fix this. The problem of parsing `0123.toString()` has fixed in https://github.com/ternjs/acorn/commit/b14cb8bea7b5fd69fb6dcf2b75cf6fb772e79ed6
